### ISSUE TITLE
MAC Problems source code

### ISSUE THREAD

**Issue by Momamari** [DARIAH-DE/TopicsExplorer#127]
Hey there, we are two students form TU Darmstadt who have problems getting to run DARIAH.

Both of us work with MacOS (Catalina 10.15.7) and we are also very new to programming, python etc.
We manage to open the topic explorer with the source code in the browser but once we chose some text files and stop words for our topic modelling the website crushes and "http://localhost:5000/modeling" is shown without any content. 
We have read the readme file but we are very insecure about how to solve this problem. So if you could help us in any way we would be really thankful!!

I guess it has something to do with the frontend/backend? But whenever I try to enter the python /usr/file/... --frozen command, I'll get:

 * Serving Flask app "topicsexplorer" (lazy loading)
 * Environment: production
   WARNING: This is a development server. Do not use it in a production deployment.
   Use a production WSGI server instead.
 * Debug mode: off
    ....
    ....
OSError: [Errno 48] Address already in use

**Comment by severinsimmler** [comment]
Hi @Momamari, thank you for your bug report. Unfortunately, this project has become somewhat dusty and is no longer actively developed. I updated a few things quick and dirty – it should run again with Python 3.8 or higher. Make sure your master branch is up-to-date and then:

1. Create a new virtual environment (`python3 -m venv env`), activate it (`source env/bin/activate`) and install the dependencies (`pip install .`)
2. Run the application with `python application.py`. Your default browser should open a new window.
3. Select your text files and start the training. Unfortunately the progress won't be displayed in the browser (this is currently broken, I might fix this over the holidays), but you can see log messages in your terminal.

Once the training is done, the results should be displayed in your browser. Hope this helps for now. Feel free to write again if anything should not work as I have described it.

**Comment by severinsimmler** [comment]
By the way, if you're already a bit into programming in Python, check out [these notebooks introducing topic modeling](https://github.com/DARIAH-DE/Topics/tree/master/notebooks).

**Comment by wissen-ist-acht** [comment]
Hi,
Even though active development has rather stopped, I wanted to report yet another issue with opening the program on a Mac:

I have been using the Topics Explorer during the last months, last time end of December 2021. My next try was middle of March 2022, and since then I have not been able to get it running. I am working with Mac OS Monterey (12.2.1).
The cause of the problem may be similar to what was reported above:
If I execute the program by clicking, I'll get this in the terminal:

 * Serving Flask app "topicsexplorer" (lazy loading)
 * Environment: production
   WARNING: Do not use the development server in a production environment.
   Use a production WSGI server instead.
 * Debug mode: off

And nothing happens. 

If I try starting the program via resources -> backend -> topics explorer and then resources -> webengine -> topics explorer, a blank browser window opens, without any option to interact.

If I start a virtual environment, a browser window opens as well, where I get:

Access to localhost was denied
You don't have authorization to view this page.
HTTP ERROR 403

I recently updated to Monterey, but tried to get the program running on a computer with OS 11.6, this did not work either. On Windows, it seems to be working, though.

If you find the time to look into this, this would be highly appreciated - while I learned in the meantime how to perform topic modeling using code, I really liked the approach of the program and its applicability for people without programming skills.

Thanks!

**Comment by severinsimmler** [comment]
Hi @wissen-ist-acht,

thank you for the bug report. This application is running on port 5000 by default. Looks like Apple introduced AirPlay Receiver with Monterey already running on port 5000 (see also [here](https://stackoverflow.com/questions/69818376/localhost5000-unavailable-in-macos-v12-monterey/69829313#69829313)). You can either turn off the AirPlay Receiver, or pull the latest changes from this repository. The DARIAH TopicsExplorer should now run on port 5001 (I have just fixed this).

Hope this helps :)
